lib/datadog/opentelemetry/metrics.rb

```diff
@@ -0,0 +1,124 @@
+          'deployment.environment' => @settings.env || '',
+          'service.version' => @settings.version || '',
+        }
+        resource_attributes['host.name'] = Datadog::Core::Environment::Socket.hostname if @settings.tracing.report_hostname
+
+        @settings.tags&.each do |key, value|
+          otel_key = case key
+          when 'service' then 'service.name'
+          when 'env' then 'deployment.environment'
+          when 'version' then 'service.version'
+          else key
+          end
+          resource_attributes[otel_key] = value
+        end
+
+        ::OpenTelemetry::SDK::Resources::Resource.create(resource_attributes)
+      end
+
+      def configure_metric_reader(provider)
+        exporter_name = @settings.opentelemetry.metrics.exporter
+        return if exporter_name == 'none'
+
+        configure_otlp_exporter(provider)
+      rescue => e
+        @logger.warn("Failed to configure OTLP metrics exporter: #{e.message}")
+      end
+
+      def resolve_metrics_endpoint
+        metrics_config = @settings.opentelemetry.metrics
+        exporter_config = @settings.opentelemetry.exporter
+
+        return metrics_config.endpoint.to_s if metrics_config.endpoint
+
+        if metrics_config.protocol
+          protocol = metrics_config.protocol
+          port = (protocol == 'http/protobuf') ? 4318 : 4317
+          path = (protocol == 'http/protobuf') ? '/v1/metrics' : ''
+          return "http://#{@agent_host}:#{port}#{path}"
+        end
+
+        return exporter_config.endpoint.to_s if exporter_config.endpoint
+
+        protocol = exporter_config.protocol || 'http/protobuf'
+        port = (protocol == 'http/protobuf') ? 4318 : 4317
+        path = (protocol == 'http/protobuf') ? '/v1/metrics' : ''
+        "http://#{@agent_host}:#{port}#{path}"
+      end
+
+      def configure_otlp_exporter(provider)
+        require 'opentelemetry/exporter/otlp_metrics'
+        require_relative 'sdk/exporter'
+
+        metrics_config = @settings.opentelemetry.metrics
+        exporter_config = @settings.opentelemetry.exporter
+        timeout = metrics_config.timeout || exporter_config.timeout
+        headers = metrics_config.headers || exporter_config.headers || {}
+
+        protocol = metrics_config.protocol || exporter_config.protocol
+        exporter = Datadog::OpenTelemetry::SDK::MetricsExporter.new(
+          endpoint: resolve_metrics_endpoint,
+          timeout: timeout / 1000.0,
+          headers: headers,
+          protocol: protocol
+        )
+
+        reader = ::OpenTelemetry::SDK::Metrics::Export::PeriodicMetricReader.new(
+          exporter: exporter,
+          export_interval_millis: metrics_config.export_interval,
+          export_timeout_millis: metrics_config.export_timeout
+        )
+        provider.add_metric_reader(reader)
+      rescue LoadError => e
+        @logger.warn("Could not load OTLP metrics exporter: #{e.message}")
+      end
```

Comment:
Maybe for these kinds of things it's not relevant, but often when errors happen in dd-trace-rb, we report them via our own telemetry. Should we do that here + other places as well?

Reply:
Yup, thats a gap in the current design. We only collect telemetry configurations and for metrics exports. We do not collect telemetry for scenarios where required packages are missing. We need to add this to the python, php, and node tracers as well